src/map/ai/controllers/mob_controller.cpp

```diff
@@ -568,40 +568,59 @@ void CMobController::DoCombatTick(time_point tick)
     HandleEnmity();
     PTarget = static_cast<CBattleEntity*>(PMob->GetEntity(PMob->GetBattleTargetID()));
 
-    if (PTarget)
+    if (TryDeaggro())
     {
-        if (TryDeaggro())
-        {
-            Disengage();
-            return;
-        }
+        Disengage();
+        return;
+    }
 
-        TryLink();
+    TryLink();
 
-        float currentDistance = distance(PMob->loc.p, PTarget->loc.p);
+    PMob->PAI->EventHandler.triggerListener("COMBAT_TICK", CLuaBaseEntity(PMob));
+    luautils::OnMobFight(PMob, PTarget);
 
-        PMob->PAI->EventHandler.triggerListener("COMBAT_TICK", CLuaBaseEntity(PMob));
-        luautils::OnMobFight(PMob, PTarget);
+    if (PMob->PAI->IsCurrentState<CInactiveState>() || !PMob->PAI->CanChangeState())
+    {
+        return;
+    }
 
-        if (PMob->PAI->IsCurrentState<CInactiveState>() || !PMob->PAI->CanChangeState())
+    if (PFollow != nullptr && (m_follow & FOLLOW_RUN_AWAY) == FOLLOW_RUN_AWAY)
+    {
+        if (distance(PMob->loc.p, PFollow->loc.p) > FollowRunAwayDistance)
         {
-            return;
+            if (!PMob->PAI->PathFind->IsFollowingPath())
+                PMob->PAI->PathFind->PathTo(PFollow->loc.p);
+            PMob->PAI->PathFind->FollowPath(m_Tick);
         }
-        else if (IsSpecialSkillReady(currentDistance) && TrySpecialSkill())
+        else
         {
-            return;
+            PMob->PAI->EventHandler.triggerListener("RUN_AWAY", CLuaBaseEntity(PMob), CLuaBaseEntity(PFollow));
```

Comment:
Can you document this in AI_Events.txt please. Not sure why CI is passing since it did pick this up...